src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -520,6 +520,18 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
                 // if any of the error fields are set, throw error null
                 if (!string.IsNullOrEmpty(authorizationResponse.Error))
                 {
+                    // Note: access_denied errors are special protocol errors indicating the user didn't
+                    // approve the authorization demand requested by the remote authorization server.
+                    // Since it's a frequent scenario (that is not caused by incorrect configuration),
+                    // access_denied errors are handled differently if AccessDeniedPath was populated.
+                    if (Options.AccessDeniedPath.HasValue &&
+                        string.Equals(authorizationResponse.Error, "access_denied", StringComparison.Ordinal))
```

Comment:
Because OIDC uses `string.Equals` while OAuth uses `StringValues.Equals`, that doesn't offer an overload accepting an explicit comparison type. I'm fine changing that if you don't like it.